Disable wide-char specific code

The vast majority of the libxmlrpc code nicely handles the absence of
wide char support, except at one location, which is fixed by this
patch.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/src/xmlrpc_decompose.c
===================================================================
--- a/src/xmlrpc_decompose.c
+++ b/src/xmlrpc_decompose.c
@@ -217,7 +217,11 @@
         xmlrpc_strfree(*decompRootP->store.Tstring.valueP);
         break;
     case 'w':
+#if HAVE_UNICODE_WCHAR
         free((void*)*decompRootP->store.TwideString.valueP);
+#else
+	XMLRPC_ASSERT(false);
+#endif
         break;
     case '6':
         free((void*)*decompRootP->store.TbitString.valueP);
